src/FluentValidation/TestHelper/ValidatorTestExtensions.cs

```diff
@@ -48,6 +48,19 @@ public static class ValidationTestExtension {
 			return testValidationResult.ShouldHaveValidationErrorFor(expression);
 		}
 
+		public static IEnumerable<ValidationFailure> ShouldHaveValidationErrorFor<T>(this IValidator<T> validator, Expression<Func<T, T>> expression, string ruleSet)
+    where T : class, new()
+        {
+            T instance = Activator.CreateInstance<T>();
+            return validator.TestValidate<T>(instance, ruleSet).ShouldHaveAnyValidationError<T>();
+        }
+
+		public static IEnumerable<ValidationFailure> ShouldHaveValidationErrorFor<T>(this IValidator<T> validator, Expression<Func<T, T>> expression, T value, string ruleSet)
+		where T : class, new()

```

Comment:
Remove the `new()` constraint - this is the overload that doesn't create the instance itself, so this constraint is too restrictive.